services/wallet/api.go

```diff
@@ -0,0 +1,30 @@
+package wallet
+
+import (
+	"context"
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// API is class with methods available over RPC.
+type API struct {
+	s *Service
+}
+
+// GetTransfers returns transfers in range of blocks. If `end` is nil all transfers from `start` will be returned.
+// TODO(dshulyak) benchmark loading many transfers from database. We can avoid json unmarshal/marshal json if we will
+// return header, tx and receipt as a raw json.
+func (api *API) GetTransfers(ctx context.Context, start, end *big.Int) ([]Transfer, error) {
+	log.Debug("call to get transfers", "start", start, "end", end)
+	if api.s.db == nil {
```

Comment:
i actually don't want to add more methods to Service. initially i wanted to pass db to API and make sure that it doesn't have access to Service at all. but then i had to create db after API is actually created

Reply:
Not sure if it matters here but generally it's better to have business logic in a service and API being just a wrapper. Then, you can use the service in Go and the API layer if going through JSON-RPC. If you put everything in the API, it might be inefficient for Go due to limitation of the API methods.

If you don't think that is the case here, let's keep it.

Reply:
i am not sure that i like when service implements all this logic. if i had to write multi-component service i would probably leave service to manage lifecycle and implement p2p interface. all other logic would be in sub-components, grouped by function for example Read and Send otr by objects that are used in each component. 

in general there is no problem using API directly in golang, one exception is when rpc.Subscription is returned. but if thats the issue there could be one more layer between service and api